### PR TITLE
Allow sequential modules in module list

### PR DESCRIPTION
**Pull request by eellison**
Fix for https://github.com/pytorch/pytorch/issues/16845

### CODE REVIEW

File: torch/jit/__init__.py [hunk 1]
```diff
@@ -1141,6 +1141,12 @@ def __setattr__(self, attr, value):
                 # parent module. To do this, we create a ConstModuleList, which is itself a module, that
                 # contains each of these modules as submodules. The ConstModuleList then
                 # is set as an attribute of the parent module.
+
+                # allow Sequential Models to nest within ModuleLists
+                for i in range(len(value)):
```
Comment (eellison) [inline torch/jit/__init__.py]: Acually this should be recursive

Comment (driazati) [inline torch/jit/__init__.py]: Could we refactor this so that this works recursively? (i.e. ModuleList of ModuleList of Sequential)

**Review by driazati — APPROVED** [review]

**Comment by sidazhang** [comment]
Weird question. Would this allow module list of module list?

File: torch/jit/__init__.py [hunk 2]
```diff
@@ -1136,14 +1136,23 @@ def __setattr__(self, attr, value):
 
             if hasattr(self, attr):
                 raise RuntimeError("attempting to re-assign constant '{}'".format(attr))
-            if isinstance(value, ModuleList):
+
+            def conv_module_to_const(module_value):
```
Comment (driazati) [inline torch/jit/__init__.py]: Can this be the whole ‘convert’? It’s confusing at first glance with the actual conv modules but they’re

**Comment by eellison** [comment]
> Weird question. Would this allow module list of module list?

Yep!

File: torch/jit/__init__.py (hunk 2 above)
Comment (eellison) [reply torch/jit/__init__.py]: what do you mean exactly?

**Pull request closed without merging**